localstack/utils/cloudformation/template_deployer.py

```diff
@@ -947,23 +973,35 @@ def retrieve_resource_details(resource_id, resource_status, resources, stack_nam
             if not result:
                 return None
             return result[0]
+
         elif resource_type == 'SecretsManager::Secret':
             secret_name = resource_props.get('Name') or resource_id
             secret_name = resolve_refs_recursively(stack_name, secret_name, resources)
             return aws_stack.connect_to_service('secretsmanager').describe_secret(SecretId=secret_name)
+
         elif resource_type == 'Elasticsearch::Domain':
             domain_name = resource_props.get('DomainName') or resource_id
             domain_name = resolve_refs_recursively(stack_name, domain_name, resources)
             return aws_stack.connect_to_service('es').describe_elasticsearch_domain(DomainName=domain_name)
+
         elif resource_type == 'KinesisFirehose::DeliveryStream':
             stream_name = resource_props.get('DeliveryStreamName') or resource_id
             stream_name = resolve_refs_recursively(stack_name, stream_name, resources)
             return aws_stack.connect_to_service('firehose').describe_delivery_stream(DeliveryStreamName=stream_name)
+
+        elif resource_type == 'IAM::Policy':
+            client = aws_stack.connect_to_service('iam')
+            ref = resource['Properties']['Roles'][0]['Ref']
```

Comment:
Please note that `Roles` is optional (i.e., may be empty). 

Also - wondering if we should use `get_policy(..)` instead of `get_role_policy(..)` below?